src/NuGet.Services.AzureSearch/DocumentUtilities.cs

```diff
@@ -0,0 +1,239 @@
+        /// an empty string. The third value is the framework that the dependency targets. If the third value is an
+        /// empty string or excluded, this means the dependency applies to any framework. If the package ID and range
+        /// and empty strings but the framework is included, this means that the package supports the specified
+        /// framework but has no dependencies specific to that framework.
+        /// 
+        /// Output:
+        ///     [DEPENDENCY[|DEPENDENCY][|...][|DEPENDENCY]]
+        /// Each dependency:
+        ///     [PACKAGE_ID]:[VERSION_RANGE][:TARGET_FRAMEWORK]
+        ///   
+        /// Example A (no target frameworks):
+        ///     Microsoft.Data.OData:5.0.2|Microsoft.WindowsAzure.ConfigurationManager:1.8.0
+        /// Example B (target frameworks):
+        ///     NETStandard.Library:1.6.1:netstandard1.0|Newtonsoft.Json:10.0.2:netstandard1.0
+        /// Example C (just target framework):
+        ///     ::net20|::net35|::net40|::net45|NETStandard.Library:1.6.1:netstandard1.0
+        /// </summary>
+        private static string GetFlattenedDependencies(PackageDetailsCatalogLeaf leaf)
+        {
+            if (leaf.DependencyGroups == null)
+            {
+                return null;
+            }
+
+            var builder = new StringBuilder();
+            foreach (var dependencyGroup in leaf.DependencyGroups)
+            {
+                var targetFramework = dependencyGroup.ParseTargetFramework();
+
+                if (dependencyGroup.Dependencies != null && dependencyGroup.Dependencies.Any())
+                {
+                    foreach (var packageDependency in dependencyGroup.Dependencies)
+                    {
+                        AddFlattennedPackageDependency(targetFramework, packageDependency, builder);
+                    }
+                }
+                else
+                {
+                    if (builder.Length > 0)
+                    {
+                        builder.Append("|");
+                    }
+
+                    builder.Append(":");
+                    AddFlattenedFrameworkDependency(targetFramework, builder);
+                }
+            }
+
+            return builder.Length > 0 ? builder.ToString() : null;
+        }
+
+        private static void AddFlattennedPackageDependency(
+            NuGetFramework targetFramework,
+            PackageDependency packageDependency,
+            StringBuilder builder)
+        {
+            if (builder.Length > 0)
+            {
+                builder.Append("|");
+            }
+
+            builder.Append(packageDependency.Id);
+            builder.Append(":");
+
+            var versionRange = packageDependency.ParseRange();
+
+            if (!VersionRange.All.Equals(versionRange))
+            {
+                builder.Append(versionRange?.ToString("S", new VersionRangeFormatter()));
```

Comment:
Can the `VersionRangeFormatter` instance be `static readonly`?

Reply:
Sure, good idea.